src/js/lib/monster.routing.js

```diff
@@ -104,7 +104,7 @@ define(function(require) {
 		},
 
 		addDefaultRoutes: function() {
-			this.add(/^apps\/([a-z]+(?:-[a-z]+)?)(\/[^?]*)?\??(.*)?$/, function(appName, restSegment, queryString) {
+			this.add(/^apps\/([a-z]+(?:-[a-z]+(-[a-z]+){0,1})?)(\/[^?]*)?\??(.*)?$/, function(appName, restSegment, queryString) {
```

Comment:
This pattern won't work for app IDs with more than three hyphened words.

How about making it generic, by making the non-catching group greedy (match as need, 0 or more times):

```suggestion
			this.add(/^apps\/([a-z]+(?:-[a-z]+)*)(\/[^?]*)?\??(.*)?$/, function(appName, restSegment, queryString) {
```
![download (1)](https://user-images.githubusercontent.com/1958193/184254911-2cb49404-af0c-4850-8d66-4c7a659ed968.png)

Reply:
Thanks, I was looking for a way to make it match 0 or more but forgot about the `*`
![image](https://user-images.githubusercontent.com/8347582/184431191-58446411-7952-4983-bd40-99ad795c4b66.png)

Done